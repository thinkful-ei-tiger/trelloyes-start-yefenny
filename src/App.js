import React from 'react';
import './App.css';

import List from './List';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      store: this.props.store
    };
  }
  newRandomCard = () => {
    const id =
      Math.random().toString(36).substring(2, 4) +
      Math.random().toString(36).substring(2, 4);
    return {
      id,
      title: `Random Card ${id}`,
      content: 'lorem ipsum'
    };
  };
  deleteCard = (id) => {
    console.log(`Delete card function with id ${id}`);
  };

  getList = (listId, cardId) => {
    let newList = [];
    let newItem = {};
    this.state.store.lists.map((item, index) => {
      if (item.id === listId) {
        newItem = item;
      } else newList.push(item);
    });
    newItem.cardIds = [...newItem.cardIds, cardId];
    newList.push(newItem);
    newList.sort((first, second) => {
      if (first.id < second.id) return -1;
      else return 1;
    });
    return newList;
  };

  addRandomCard = (listId) => {
    const newCard = this.newRandomCard();
    const name = newCard.id;
    const newList = this.getList(listId, name);

    this.setState((prevState) => ({
      store: {
        allCards: {
          ...prevState.store.allCards,
          [name]: newCard
        },
        lists: newList
      }
    }));
  };

  render() {
    return (
      <main className='App'>
        <header className='App-header'>
          <h1>Trelloyes!</h1>
        </header>
        <div className='App-list'>
          {this.state.store.lists.map((list) => {
            let cards = list.cardIds.map((x) => this.state.store.allCards[x]);
            return (
              <List
                key={list.id}
                id={list.id}
                header={list.header}
                cards={cards}
                deleteCard={this.deleteCard}
                addRandomCard={this.addRandomCard}
              />
            );
          })}
        </div>
      </main>
    );
  }
}

export default App;
