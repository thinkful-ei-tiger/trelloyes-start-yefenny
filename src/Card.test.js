import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from './Card';
import App from './App';

describe('Card Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI with props as expected', () => {
    const wrapper = shallow(
      <Card
        title='Card 1'
        content='My content'
        id='a'
        deleteCard={App.deleteCard}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
