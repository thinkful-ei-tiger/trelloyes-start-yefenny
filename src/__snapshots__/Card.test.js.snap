// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the UI with props as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
  <h3>
    Card 1
  </h3>
  <p>
    My content
  </p>
</div>
`;
